### PR TITLE
Just compare what is necessary for equality.

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR
@geirst FYI

### CODE REVIEW

**Comment by baldersheim** [comment]
@vekterli Ping...

File: document/src/main/java/com/yahoo/document/ReferenceDataType.java [hunk 1]
```diff
@@ -98,4 +98,19 @@ public boolean isValueCompatible(FieldValue value) {
         ReferenceFieldValue rhs = (ReferenceFieldValue)value;
         return rhs.getDataType().equals(this);
     }
+
+    @Override
+    public int compareTo(DataType rhs) {
+        int cmp = super.compareTo(rhs);
+        return (cmp != 0)
+                ? cmp
+                : (rhs instanceof ReferenceDataType) ? targetType.compareTo(((ReferenceDataType) rhs).targetType) : 0;
```
Comment (vekterli) [inline document/src/main/java/com/yahoo/document/ReferenceDataType.java]: Consider adding parenthesis around the nested ternary expression to make precedence more obvious.

**Review by vekterli — APPROVED** [review]
:+1:

File: document/src/main/java/com/yahoo/document/ReferenceDataType.java (hunk 1 above)
Comment (baldersheim) [reply document/src/main/java/com/yahoo/document/ReferenceDataType.java]: Fixed

**Pull request merged**